website/docs/components/form/key-value-inputs/partials/guidelines/guidelines.md

```diff
@@ -0,0 +1,132 @@
+
+![At the bottom of the input row is a neutral alert that says "Limit of four invites has been reached."](/assets/components/form/key-value-inputs/key-value-inputs-row-limit.png)
+
+If the maximum number of rows has been reached and the user deletes a row, the alert should be replaced by the add button.
+
+### Deleting rows
+
+If more than one row is present, users can opt to delete any of the rows.
+
+![A user's cursor is hovering over the second row's delete button](/assets/components/form/key-value-inputs/key-value-inputs-ip-delete-any-row.png)
+
+A single row is required at all times. A delete button will only appear for the first row if a second row has been added.
+
+![A single row with a key and value input with no delete button on the side.](/assets/components/form/key-value-inputs/key-value-inputs-single-row.png)
+
+## Footer
+
+The footer supports an add row button and an alert.
+
+### Add row button
+
+The add row button is displayed by default and appends rows to collect additional data and information. The labeling of this button follows our content guidelines of [“add and remove.”](/content/writing-style#add-and-remove)
+
+![A single secondary button with a leading plus icon with the label "Add user"](/assets/components/form/key-value-inputs/key-value-inputs-footer-add-button.png)
+
+### Alert
+
+The alert is used to notify the user of a max row limit being hit. The messaging of this alert follows the language paradigm of “Limit of {max number if word format} {object} has been reached.” 
+
+This uses the compact neutral alert style so it doesn’t visually compete with any other higher priority validation feedback, like input errors.
+
+![A compact neutral alert that says "Limit of fifteen invites has been reached."](/assets/components/form/key-value-inputs/key-value-inputs-footer-notification.png)
+
+## Responsive behavior
+
+In experiences that are greater than or equal to the [`sm` breakpoint](/foundations/breakpoints), a visible label and helper text are only present in the first row. This is because the form elements are condensed which will help reduce visual clutter.
+
+![Key value input rows with the key and value are side by side along with a delete button to the right. Only one visible label and helper text per grouping of key and value inputs. Add new row button below the group of inputs.](/assets/components/form/key-value-inputs/key-value-inputs-ip-address-desktop.png)
+
+However, when the viewport is less than the `sm` breakpoint, all field elements stack and occupy 100% of the container and display the label and helper text for their associated inputs. Since the form height increases, the visible labels and helper text orient the user as they are scrolling down and filling in information. If repeating each input’s helper text isn’t useful to the user, consider putting that information in the fieldset helper text instead to reduce the visual noise.
```

Comment:
Reading isn't always linear, so it's best to link in both places. 

```suggestion
However, when the viewport is less than the [`sm` breakpoint](/foundations/breakpoints), all field elements stack and occupy 100% of the container and display the label and helper text for their associated inputs. Since the form height increases, the visible labels and helper text orient the user as they are scrolling down and filling in information. If repeating each input’s helper text isn’t useful to the user, consider putting that information in the fieldset helper text instead to reduce the visual noise.
```